docs/flags.md

```diff
@@ -0,0 +1,130 @@
+*) Describe how this is implemented. These details, especially around persistence, will let you understand how to write tooling that interacts with this system.
+*) Future work (we see how it fits in but don't plan to implement until there's concrete demand).
+
+## Examples
+
+### Run only some services
+Your app has many services, call them A, B, C, and D. You want users to be able to run only some of them.
+
+and you want your users to be able to run only a subset of them. E.g., `tilt up` will run all services, but running `tilt up a b d` will run only the resources a, b, and d, ignoring c. Each run of Tilt will start from scratch.
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up a b d`: run A, B, and D but not C
+* `tilt flags a b`: change running Tilt to run A and B but neither C nor D.
+* `tilt up` (a second time): run all services
+
+#### Tiltfile
+```python
+flags.define_string_list("to-run", args=True, sticky=False)
+cfg = flags.parse()
+
+flags.exclude_all_resources()
+for r in cfg['to-run']:
+  flags.include_resource(r)
+```
+
+### Run a defined set of services
+Your app has many services, call them A, B, C, and D. You have defined modes you want users to be able to start (e.g. consumer vs. enterprise) and you don't want them to have to remember which service is necessary for which mode.
+
+You want to be able to write a doc that tells users to run `tilt up consumer` the first time and know that's enough that they'll be set up. Running Tilt subsequent times without arguments should default to the mode they already have set.
+
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up consumer`: run consumer services and record mode
+* `tilt up` (a second time): run consumer services
+
+#### Tiltfile
+```python
+flags.define_string_list("to-run", args=True)
+cfg = flags.parse()
+
+groups = {
+  'consumer': ['a', 'b', 'c'],
+  'enterprise': ['a', 'b', 'd'],
+}
+
+
+
+
+
+
+
+
+
+
+
+
+flags.exclude_all_resources()
+for arg in cfg['args']:
+  if arg in groups:
+    for r in groups[arg]:
+      flags.include_resource(r)
+  else:
+    flags.include_resource(r)
+```
+
+By making the flag sticky (the default), Tilt will record the flag between runs.
+
+
+### Specify services to edit
+You have many services that your users frequently need to run, but don't expect to edit. Your YAML includes images from your CI that are suitably recently (and match what's in production or staging).
+
+#### Command-Lines
+* `tilt up`: run all services, editing none
+* `tilt up -- --to-edit b`: run all services, editing b
+* `tilt up -- consumer --to-edit b --to-edit c`: run consumer services and setup b and c for editing.
+
+#### Tiltfile
+```python
+flags.define_string_list("to-run", args=True)
+flags.define_string_list("to-edit")
+cfg = flags.parse()
+
+# omitting the code from above Tiltfile that includes correct resources
+
+# only configure the build for services we want to edit
```

Comment:
I haven't put too much thought into the wording, but it might be worth explicitly calling out something like "Thanks to Tilt's existing behavior, images that don't have a `docker_build` directive will just deploy with whatever is configured in their yaml, so it's really just this easy"

Reply:
Yep; will incorporate.